### PR TITLE
3987 test with factories

### PR DESCRIPTION
**Pull request by easherma**
# Description

# Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How can this be tested?

 

# Checklist:
- [ ] Request reviewers
- [ ] Slack-out message for review
- [ ] Link this PR in the issue
- [ ] Moved card to *review* in zenhub

### CODE REVIEW

File: README/testing.md [hunk 1]
```diff
@@ -17,6 +17,40 @@ In the case of publish_preflight tests, we needed to create mock forms to valida
...
+### Using Factories to make test data in the database
+
+The most basic way would be to:
+1. have joplin running,
+2. go into the shell like
+`pipenv run ./joplin/manage.py shell_plus`
+(or just 'shell')
+3. Import your factory:
+`from locations.factories import LocationPageFactory`
+`from base.locations import ServicePageFactory`
```
Comment (briaguya) [inline README/testing.md]: this line fails

Comment (easherma) [reply README/testing.md]: thx, corrected!

File: joplin/base/factories/guide_page.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import factory.fuzzy
+import wagtail_factories
+from django.utils.text import slugify
+from wagtail.core.models import Collection, Page
+from base.models import *
+from wagtail.core import blocks
+from .. import PageFactory, TextBlockFactory
+
+"""
+almost working
```
Comment (easherma) [inline joplin/base/factories/guide_page.py]: this does work, but it atm relies on some existing objects in the database that I'd like to eventually also replace with factories

There is some goofiness due to M2M relationships with Clusterable Model, though

File: joplin/base/tests/test_information_page.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
+from base.models import InformationPageContact
+from base.factories.information_page import InformationPageContactFactory, InformationPageFactory
+import pytest
+
+
+@pytest.mark.django_db
```
Comment (easherma) [inline joplin/base/tests/test_information_page.py]: this decorator means 'i need the db for this test' 
Thats it! no setup, teardown code required

File: joplin/base/tests/test_information_page.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+def test_information_page_factory(information_page_factory):
```
Comment (easherma) [inline joplin/base/tests/test_information_page.py]: this passes the factory as a fixture, so you don't need to import it again

File: joplin/base/views/content_list_view_publish.py [hunk 5]
```diff
@@ -80,16 +82,17 @@ def publish(request, page_id):
...
 # in order to add a "Publish" option to the Content List View's "more" dropdown
-if page_perms.can_publish() and page.has_unpublished_changes:
-    yield Button(
-        _('Publish'),
-        reverse('publish', args=[page.id]),
-        attrs={'title': _("Publish page '{title}'").format(
-            title=page.get_admin_display_title())},
-        priority=50
-    )
+# if page_perms.can_publish() and page.has_unpublished_changes:
```
Comment (easherma) [inline joplin/base/views/content_list_view_publish.py]: @Niicck didn't we get rid of this functionality?

**Pull request merged**